plugins/outputs/mqtt/template_parser.go

```diff
@@ -0,0 +1,47 @@
+package mqtt
+
+import (
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+type TemplateTopic struct {
+	Hostname    string
+	metric      telegraf.Metric
+	topicPrefix string
+}
+
+func (t *TemplateTopic) Tag(key string) string {
+	tagString, _ := t.metric.GetTag(key)
+	return tagString
+}
+
+func (t *TemplateTopic) TopicPrefix() string {
+	return t.topicPrefix
+}
+
+func (t *TemplateTopic) PluginName() string {
+	return t.metric.Name()
+}
+
+func (t *TemplateTopic) Parse(m *MQTT) string {
```

Comment:
With the above change to the struct this becomes
```suggestion
func (t *TemplateTopic) Parse(m telegraf.Metric) string {
```